internal/controller/prefectserver_controller.go

```diff
@@ -270,20 +267,26 @@ func (r *PrefectServerReconciler) reconcileDeployment(ctx context.Context, serve
 
 		condition = conditions.Updated(objName)
 	} else {
-		imageVersion := prefectiov1.VersionFromImage(desiredDeployment.Spec.Template.Spec.Containers[0].Image)
-		if server.Status.Version != imageVersion ||
-			!meta.IsStatusConditionTrue(server.Status.Conditions, "DeploymentReconciled") {
+		if !meta.IsStatusConditionTrue(server.Status.Conditions, "DeploymentReconciled") {
+			condition = conditions.Updated(objName)
+		}
 
-			server.Status.Version = imageVersion
+		ready := foundDeployment.Status.AvailableReplicas > 0
```

Comment:
For example, [here is how the Kubernetes Deployment controller calculates if a Deployment is complete](https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/deployment/util/deployment_util.go#L706-L713):

```go
// DeploymentComplete considers a deployment to be complete once all of its desired replicas
// are updated and available, and no old pods are running.
func DeploymentComplete(deployment *apps.Deployment, newStatus *apps.DeploymentStatus) bool {
	return newStatus.UpdatedReplicas == *(deployment.Spec.Replicas) &&
		newStatus.Replicas == *(deployment.Spec.Replicas) &&
		newStatus.AvailableReplicas == *(deployment.Spec.Replicas) &&
		newStatus.ObservedGeneration >= deployment.Generation
}
```

Reply:
Great question!  What I was hoping for here is a simple boolean signal telling us whether the `PrefectServer` is ready to accept traffic, which would be the case if there's at least 1 available replica.  During an upgrade, we might go up to 2 and then back down to 1, and during that time we're still available and ready to receive requests.  I think that's what we want here.

But now you've got me thinking about how the `Status.Version` fields work, which is looking at the Deployment and not the underlying Pods.  The truth is that you could have multiple versions running in parallel during an upgrade, so maybe that should be a list like:

```
status: [
  {
    version: "3.1.2",
    ready: 2
  },
  {
    version: "3.1.3",
    ready: 1
  },
]
```

This might be useful to understand during a version bump, not sure yet...